src/AutoRest.CSharp/Input/CodeModelPartials.cs

```diff
@@ -147,6 +161,7 @@ internal partial class Schema
     {
         public string? XmlName => Serialization?.Xml?.Name;
         public string Name => Language.Default.Name;
+        public string? NameOverride;
```

Comment:
We already have multiple ways to override schema name (x-ms-clientName, using [CodeGenProperty]). I'm concerned that customer would get slightly confuset about which way should they use. Can we add a swagger directive that would feed into x-ms-client-name extension  that we already support?

Reply:
we didn't want to disrupt that flow since my understanding is that will affect all languages which we don't want to do

Reply:
We will need to investigate if this flag behaves in the way we need it to.  If so we will convert over I've opened up an ADO to track this moving forward https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5746